### PR TITLE
Locomotive radio control

### PR DESCRIPTION
**Pull request by Endalion**
Added a new item that has a similar component as LocoControlAugment but is able to directly access a train from its UUID. Currently manifests itself as an OC card; you use it in a computer case/adapter/server.

To link to a locomotive, right click the loco with the card in hand. To unlink, shift right-click any locomotive with the card in hand.

Content list
- [x] Recipe
- [x] Unique texture/model
- [x] Limited Range

### CODE REVIEW

**Comment by FriedrichLP** [comment]
Your code looks quite good but you used a lot of wrong java syntax. Please look over your code, use a uniform syntax and correct the existing one. One Example: You placed the brackets on the next line sometimes, which is ok but I'd reccomend putting them on the same line because it's like that in most of IR's code. I can list every syntax issue if you'd like. But you still did a good job!

**Comment by FriedrichLP** [comment]
In Locomotive.java and RadioCtrlCardDriver.java are still some unnecessary lines. In ItemRadioCtrlCard.java are some tabs incorrect (line 23 and 24) and the if statement in addInformation() isnt correct. You can also put the ".equals(UUID.fromString(stack.getTagCompound().getString("linked_uuid"))))" from the most recent changes into the same line as the if statement and the whole for loop inside that method has one tab to much. You could also add brackets to the if statement inside RadioCardCtrlDriver.java#isWork
The RadioCardCtrlDriver.java#normalize method also has some unnecessary lines
After those changes and after cam reviewed your pull request it should be ready to be merged

**Comment by cam72cam** [comment]
Thanks for submitting this PR, it's been on my TODO list for a while!

Once we get it buttoned up I'll be happy to merge it.

I'll try to do an in depth review tonight after work.

**Comment by Endalion** [comment]
> Thanks for submitting this PR, it's been on my TODO list for a while!
> 
> Once we get it buttoned up I'll be happy to merge it.
> 
> I'll try to do an in depth review tonight after work.

No problem, I had spare time and was really itching for some automatic train networks xD

File: src/main/java/cam72cam/immersiverailroading/items/ItemRadioCtrlCard.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+public class ItemRadioCtrlCard extends Item {
+	public static final String NAME = "item_remote_card";
+	
+	public ItemRadioCtrlCard() {
+		super();
+		setUnlocalizedName(ImmersiveRailroading.MODID + ":" + NAME);
+		setRegistryName(new ResourceLocation(ImmersiveRailroading.MODID, NAME));
+        this.setCreativeTab(ItemTabs.MAIN_TAB);
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/items/ItemRadioCtrlCard.java]: Please fix the indentation here

**Comment by cam72cam** [comment]
Once a recipe and item model (2d is fine) are added, this should be all set to be merged!

**Comment by cam72cam** [comment]
If you need help with the texture ping @modeler in discord.  I'd like to get this in by the weekend if possible.

File: src/main/java/cam72cam/immersiverailroading/items/ItemRadioCtrlCard.java (hunk 1 above)
Comment (Endalion) [reply src/main/java/cam72cam/immersiverailroading/items/ItemRadioCtrlCard.java]: Fixed in 41849dc

File: src/main/java/cam72cam/immersiverailroading/thirdparty/opencomputers/RadioCtrlCardDriver.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		@Callback(doc = "function(double) -- sets the locomotive throttle")
+		public Object[] setThrottle(Context context, Arguments arguments) throws Exception {
+			if (radioDrain()) {
+				linkedLoco.setThrottle(normalize(arguments.checkDouble(0)));
+			}
+			return null;
+		}
+
+		private float normalize(double val) {
+			if (val > 1) {
```
Comment (cam72cam) [inline src/main/java/cam72cam/immersiverailroading/thirdparty/opencomputers/RadioCtrlCardDriver.java]: Add a Double.isNan check here.  I added that to the other normalize fn

**Comment by Endalion** [comment]
It should be done now, apologies for the late updates.

**Pull request merged**